### ISSUE TITLE
Enable `cypress/recommended`

### ISSUE THREAD

**Issue by menelaos** [tools-aoeur/eslint-config#4]
[`eslint-plugin-cypress`](https://github.com/cypress-io/eslint-plugin-cypress) provides `recommended` rules. Investigate whether it makes sense to enable all of them.

See https://github.com/tools-aoeur/eslint-config/pull/3#discussion_r1807051400.

**Comment by amannn** [comment]
There was some previous discussion on this here: https://github.com/tools-aoeur/eslint-config/pull/3#discussion_r1807051400. What are problematic cases you're currently running into where you'd wish the linter provided more support?

**Comment by jdehaan** [comment]
avoid `wait`, avoid chaining in inappropriate ways... I had to activate the eslint-plugin-cypress with the recommended rules myself isnide horusnome, somehow these rules are not active despite having selected the `cypress` feature flag for this package. Maybe I did something wrong but it looks to me the cypress recommended rules are not effective solely by using this configuration package

**Comment by menelaos** [comment]
The `cypress` feature flag for this package only [activates](https://github.com/tools-aoeur/eslint-config/blob/79238827ff254ce4a7ebf0d6140b4c66d2cc99f8/src/cypress.js#L12) the [`no-restricted-properties`](https://eslint.org/docs/latest/rules/no-restricted-properties) rule currently. The background can be found in the [discussion](https://github.com/tools-aoeur/eslint-config/pull/3#discussion_r1807051400) that @amannn mentioned.

**Comment by jdehaan** [comment]
Oh thx for precision... I begin to understand... But I am still wondering there is no cypress in the package.json https://github.com/tools-aoeur/eslint-config/blob/79238827ff254ce4a7ebf0d6140b4c66d2cc99f8/package.json#L35 and yet things are being done. I understand and the defaults are meaningful but yet it seems I had to install the cypress plugin explicitly myself as a dependency of the sourcing project to make things work. Maybe I'm totally missing the point but I thought eslint.config should also bring in these dependencies and have the mission to ensure these all play well together while allowing users to mute this to that feature range that is not needed in the project using the config. Am I also wrong with that?

**Comment by menelaos** [comment]
> Ahh now I understand also finally the rest, the added check was serviced by the base functionality of eslint and a config.... 

Yes, exactly. It becomes `cypress`-related due to checking for `it`, `describe` and `test`: https://github.com/tools-aoeur/eslint-config/blob/79238827ff254ce4a7ebf0d6140b4c66d2cc99f8/src/cypress.js#L14

Side note: [Aliases](https://docs.cypress.io/app/core-concepts/writing-and-organizing-tests#Test-Structure) like `context` and `specify` would currently be missed by this rule. The official [documentation](https://github.com/cypress-io/eslint-plugin-cypress?tab=readme-ov-file#mocha-only-and-skip) mentions [`mocha/no-exclusive-tests`](https://github.com/lo1tuma/eslint-plugin-mocha/blob/main/docs/rules/no-exclusive-tests.md) for this use case. Maybe something to look into...
